cmd/view-fits/main.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015 The astrogo Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"log"
+	"os"
+
+	"github.com/astrogo/fitsio"
+
+	"golang.org/x/exp/shiny/driver"
+	"golang.org/x/exp/shiny/screen"
+	"golang.org/x/mobile/event/key"
+	"golang.org/x/mobile/event/paint"
+)
+
+var (
+	hduArray []fitsio.Image
+)
+
+func init() {
+
+	var inputFileName string
+	var inputFile *os.File
+	var fitsFile *fitsio.File
+	var err error
+
+	if len(os.Args) < 2 {
+		log.Fatal("no image file specified")
+	}
+
+	// Getting the file argument.
+	inputFileName = os.Args[1]
+
+	// Opening the file.
+	if inputFile, err = os.Open(inputFileName); err != nil {
+		log.Fatal("Can not open the input file: %s.", err)
+	}
+
+	// Opening the FITS file.
+	if fitsFile, err = fitsio.Open(inputFile); err != nil {
+		log.Fatal("Can not open the FITS input file: %s.", err)
+	}
+
+	// Getting the file HDUs.
+	HDUs := fitsFile.HDUs()
+	for _, HDU := range HDUs {
+		// Getting the header informations.
+		header := HDU.Header()
+		axes := header.Axes()
+		log.Printf("axes:%v", axes)
+
+		// Leaving HDU with no axes.
+		if len(axes) != 0 {
+			// Keeping HDU with IMAGE_HDU type.
+			if HDUType := HDU.Type(); HDUType == fitsio.IMAGE_HDU {
+				// Filling the HDU array.
+				hduArray = append(hduArray, HDU.(fitsio.Image))
+
+			}
+
+		}
+	}
+
+}
+
+func main() {
+
+	driver.Main(func(s screen.Screen) {
+
+		nbImages := len(hduArray)
+
+		if nbImages == 0 {
+			log.Fatal("No images to display.")
+		}
+		log.Println("Number of images:", nbImages)
+
+		// Current displayed image.
+		index := 0
+		// Getting the first image.
+		currentHDU := hduArray[index]
+		// Getting the axes
+		axes := currentHDU.Header().Axes()
+		width := axes[0]
+		height := axes[1]
+
+		// Building the main window.
+		w, err := s.NewWindow(&screen.NewWindowOptions{
+			Width:  width,
+			Height: height,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer w.Release()
+		// Building the screen buffer.
+		b, err := s.NewBuffer(image.Point{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer b.Release()
+
+		w.Fill(b.Bounds(), color.Black, draw.Src)
+		w.Publish()
+
+		for {
+			e := w.NextEvent()
+
+			switch e := e.(type) {
+			default:
+			case key.Event:
+				switch e.Code {
+				case key.CodeEscape, key.CodeQ:
+					return
+				case key.CodeRightArrow:
+					if e.Direction == key.DirPress {
+						if index < nbImages-1 {
+							index++
+						} else {
+							index = 0
+						}
+						b.Release()
+						w.Send(paint.Event{})
+					}
+				case key.CodeLeftArrow:
+					if e.Direction == key.DirPress {
+						if index == 0 {
+							index = nbImages - 1
+						} else {
+							index--
+						}
+						b.Release()
+						w.Send(paint.Event{})
+					}
+				}
+			case paint.Event:
+				log.Println("index:", index)
+
+				currentHDU := hduArray[index]
```

Comment:
this looks buggy. you seem to shadow the previously declared `currentHDU` variable (declared at line 86).

also, `currentHDU` could be renamed as just `hdu`.
ditto for `currentImage -> img`.